js-project-guidelines.md

```diff
@@ -11,30 +13,144 @@ The creation of this document came as a result of the hurdles that the IPFS comm
+
+#### c) Build
+
+There are a lot of build systems out there, and we have tested our fair share of them
+in the past. They all have their strength and weaknesses and at the
+end of the day it matters that the tool does what you want it to do. For
+that reason we settled on using [webpack](http://webpack.github.io/) as it gives us a large control over
+every detail when bundling and we feel quite comfortable with it at this
+point.
+
+As we are using some features of ES2015 we also need something to ensure
+our code runs on the platforms that do not have full support yet. For this
+[babel](http://babeljs.io/) is the tool of our choice.
+
+When others consume our code though, we don’t want to enforce these choices
+on them. So that’s why our build process in dignified.js creates multiple
+versions for (hopefully) everyone to use.
+
+- __Raw ES2015 version__, ready to be consumed by platforms that understand Node.js based require and most of ES2015.
+- __Raw ES5 version__, ready to be consumed by platforms that understand Node.js based require and ES5.
+- __Concatenated ES5 version__, ready to be consumed by browsers through a script tag, where size does not matter.
+- __Concatenated and minified ES5 version__, ready to be consumed by browsers through a script tag, where size matters.
+
+#### d) Release
+
+Releasing a new version entails
+
+1. Run linting
+2. Run all tests
+3. Build all three different versions
+4. Bump the version in package.json
+5. Commit the version bump
+6. Create a git tag
+7. Push to github
+8. Publish to npm
+
+### Day to day operations...
+
+#### ...for maintainers
+
+##### Setting up `dignified.js`
+
+There are a couple of binaries that dignified.js provides for you to use
+
+```sh
+$ dignified-lint
+$ dignified-test
+$ dignified-test browser
+$ dignified-test node
+$ dignified-build
+$ dignified-release major
+$ dignified-release minor
+$ dignified-release
+```
+
+these can be setup in your package.json as npm scripts:
+
+```json
+{
+  "scripts": {
+    "lint": "dignified-lint",
+    "build": "dignified-build",
+    "test": "dignified-test",
+    "test:node": "dignified-test node",
+    "test:browser": "dignified-test browser",
+    "release": "dignified-release"
+  }
+}
+```
+
+and you also need to add it your `devDependencies` by running
+
+```sh
+$ npm install --save-dev dignified.js
+```
+
+##### Directory Structure
+
+To reduce the amount of configuration dignified.js expects your source code to be in the src and
+your test files in the test directory.
 
 ```sh
-.
 ├── dist # auto-generated by the transpile and minification task.
-│   ├── browser.js
-│   └── browser.min.js
-├── lib  # auto-generated source tree, transpiled using babel. Makes the code es5 compatible/
-│   ├── index.js
-│   └── ...
-├── src  # source code. Can use the latest features (es2015(6)/es2016(7)) in JavaScript.
-│   ├── index.js
-│   └── ...
-├── tests # testes folder
+│   ├── index.js
```

Comment:
Could we make the name of these files be the name of the package? This way, if someone is using two dependencies through a `<script>`, they don't have to change the names manually.